### PR TITLE
[#792] improvement(core): Optimize the deletion process of entities that have large sub-entities

### PR DESCRIPTION
**Pull request by yuqi1129**
### What changes were proposed in this pull request?

When removing an entity, we will delete the name-id mapping instead of the entity itself. 

### Why are the changes needed?

Deleting an entity with many sub-entities is time-consuming. By removing the name-id mapping, we can delete an entity more efficiently. 

Fix: #792 

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Existing UT `testEntityDelete` can cover this change.

### CODE REVIEW

File: core/src/main/java/com/datastrato/gravitino/storage/kv/KvEntityStore.java [hunk 1]
```diff
@@ -280,20 +280,18 @@ public boolean delete(NameIdentifier ident, EntityType entityType, boolean casca
...
-          deleteAuthorizationEntitiesIfNecessary(ident, entityType);
           return transactionalKvBackend.delete(dataKey);
         });
   }
 
+  private void unbindNameAndId(NameIdentifier ident, EntityType entityType) throws IOException {
+    String identNameToIdKey = generateKeyForMapping(ident, entityType, nameMappingService);
+    nameMappingService.unbindNameAndId(identNameToIdKey);
+  }
+
```
Comment (diqiu50) [inline core/src/main/java/com/datastrato/gravitino/storage/kv/KvEntityStore.java:294]: Should we still delete these entries?
`unbindNameAndId `is the same of deleteRange(prefix_min, prefix_max)

Comment (yuqi1129) [reply core/src/main/java/com/datastrato/gravitino/storage/kv/KvEntityStore.java:294]: > Should we still delete these entries?

No, we can't use a background thread to do the work just as JVM GC

> unbindNameAndId is the same of deleteRange(prefix_min, prefix_max)

`unbindNameAndId` will make the entity and it's sub-entities inaccessible to users and they are equal logically.

**Review by diqiu50 — APPROVED** [review]
LGTM

**Comment by jerryshao** [comment]
Is it still needed to merge? @yuqi1129

**Comment by yuqi1129** [comment]
> Is it still needed to merge? @yuqi1129 

Yeah, I believe this optimization is meaningful.

File: core/src/main/java/com/datastrato/gravitino/storage/kv/KvEntityStore.java (hunk 1 above)
Comment (jerryshao) [reply core/src/main/java/com/datastrato/gravitino/storage/kv/KvEntityStore.java:294]: Will it be possible that other same-name entity still use this name?

Comment (yuqi1129) [reply core/src/main/java/com/datastrato/gravitino/storage/kv/KvEntityStore.java:294]: Not possible as each name is attched with context, for example, the key '1/2/name' represents a specific entity under a metalake (1) and catalog (2).

Comment (jerryshao) [reply core/src/main/java/com/datastrato/gravitino/storage/kv/KvEntityStore.java:294]: Do you need more tests to cover?

**Review by jerryshao — APPROVED** [review]

**Pull request merged**